build.gradle

```diff
@@ -301,6 +301,24 @@ task setupIntellijStaticAnalysis {
     dependsOn copyStylelintConfiguration, copyIntelliJProjectPluginsSettings
 }
 
+// IntelliJ project static analysis plugins version sync
+task syncIntelliJCheckStyleVersion {
+    description 'Syncs the CheckStyle version used in IntelliJ IDEA automatically to the same with the buildscript'
+    group intellijSetupGroup
+    if (file('.idea/checkstyle-idea.xml').exists()) {
```

Comment:
Any alert to users if they haven't setup checkstyle in IntelliJ?

Reply:
Because this runs automatically every time the build script changes, so I was thinking it should work behind the scenes and not show the user anything.

Is there something you have in mind that I'm missing out?

Reply:
> Because this runs automatically every time the build script changes, ...
How do you do automatically execution?

Reply:
I'm sorry I provided the description in the commit:
> This works by the task executing during configuration phase, which IntelliJ
IDEA automatically runs when changes in build script is detected (only
when `auto-import` is enabled).